packages/imperative/src/security/src/DefaultCredentialManager.ts

```diff
@@ -120,9 +120,7 @@ export class DefaultCredentialManager extends AbstractCredentialManager {
             // our calling CLI. Since our caller must supply keytar, we search for keytar
             // within our caller's path.
             const requireOpts: any = {};
-            if (require.main.filename != null) {
-                requireOpts.paths = [require.main.filename, ...require.resolve.paths("@zowe/secrets-for-zowe-sdk")];
-            }
+            requireOpts.paths = [require.main?.filename, ...require.resolve.paths("@zowe/secrets-for-zowe-sdk")].filter(Boolean);
```

Comment:
Technically this changes the original behavior so that we always pass in a list of paths to `require.resolve`, but I don't think this change should be impactful 😋

Reply:
This is a good point - I don't want to make any assumptions, and its possible that the default module resolution is special for certain projects. I've updated this to add optional chaining to the `if` check so that we only apply the paths if a filename is provided in `require.main` 😋